host/src/cli.rs

```diff
@@ -0,0 +1,176 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::fmt;
+use std::path::PathBuf;
+
+use clap::ValueEnum;
+
+#[derive(clap::Parser, Debug, Clone)]
+#[command(name = "zeth")]
+#[command(bin_name = "zeth")]
+#[command(author, version, about, long_about = None)]
+pub enum Cli {
+    /// Build blocks only on the host
+    Build(BuildArgs),
+    /// Run the block building inside the executor
+    Run(RunArgs),
+    /// Provably build blocks inside the zkVM
+    Prove(ProveArgs),
+    /// Verify a block building receipt
+    Verify(VerifyArgs),
+}
+
+impl Cli {
+    pub fn build_args(&self) -> &BuildArgs {
+        match &self {
+            Cli::Build(build_args) => build_args,
+            Cli::Run(run_args) => &run_args.build_args,
+            Cli::Prove(prove_args) => &prove_args.run_args.build_args,
+            Cli::Verify(..) => unimplemented!(),
+        }
+    }
+
+    /// Generate a unique tag for the command execution
+    pub fn execution_tag(&self) -> String {
+        let time = std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap();
+        match &self {
+            Cli::Build(args) => format!("{}_build_{}", time.as_secs(), args.tag()),
+            Cli::Run(args) => format!("{}_run_{}", time.as_secs(), args.tag()),
+            Cli::Prove(args) => format!("{}_prove_{}", time.as_secs(), args.tag()),
+            Cli::Verify(..) => unimplemented!(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, clap::ValueEnum)]
+pub enum Network {
+    /// Ethereum Mainnet
+    Ethereum,
+    /// Optimism Mainnet
+    Optimism,
+    /// Optimism Mainnet as derived from the Ethereum Mainnet
+    OptimismDerived,
+}
+
+impl fmt::Display for Network {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // use the name of the clap::ValueEnum
+        let val = self.to_possible_value().unwrap();
+        write!(f, "{}", val.get_name())
+    }
+}
+
+trait Tag {
+    fn tag(&self) -> String;
+}
+
+#[derive(clap::Args, Debug, Clone)]
+pub struct BuildArgs {
+    #[clap(
+        short = 'w',
+        long,
+        require_equals = true,
+        value_enum,
+        default_value_t = Network::Ethereum
+    )]
+    /// Network name
+    pub network: Network,
+
+    #[clap(short, long, require_equals = true)]
+    /// URL of the Ethereum RPC node
+    pub eth_rpc_url: Option<String>,
+
+    #[clap(short, long, require_equals = true)]
+    /// URL of the Optimism RPC node
+    pub op_rpc_url: Option<String>,
+
+    #[clap(short, long, require_equals = true, num_args = 0..=1, default_missing_value = "host/testdata")]
```

Comment:
Changed this back to `host/testdata`:
- the previous folder did not exist, so we got a very cryptic panic when running with the defaults
- this allows the user to run the examples from the Readme without needing an rpc API key

Reply:
changed it back to `cache_rpc` making sure that parent directories are created and proper error messages are generated.